impeller/aiks/canvas.cc

```diff
@@ -838,8 +840,18 @@ void Canvas::SaveLayer(const Paint& paint,
                        std::optional<Rect> bounds,
                        const std::shared_ptr<ImageFilter>& backdrop_filter,
                        ContentBoundsPromise bounds_promise,
-                       uint32_t total_content_depth) {
+                       uint32_t total_content_depth,
+                       bool can_distribute_opacity) {
   TRACE_EVENT0("flutter", "Canvas::saveLayer");
+  if (can_distribute_opacity) {
+    FML_LOG(ERROR) << "Distribute opacity";
+    Save(false, total_content_depth, paint.blend_mode, backdrop_filter);
+    transform_stack_.back().distributed_opacity *= paint.color.alpha;
+    return;
+  } else {
+    FML_LOG(ERROR) << "no dist opacity";
```

Comment:
@flar try as I might, I can't seem to get this to fire on any of the opacity peephole tests in the macrobenchmarks app. It seems like can_distribute_opacity is always false with impeller?

This is just running with the regular canvas.

Reply:
Can't get which to fire? This is the "false" part of the if. Does this mean it is always true?

Reply:
I mean its never true

Reply:
It's working for the dl_sk_dispatcher, though...?

Reply:
(or we would have had benchmark regressions).

Reply:
but its not working here :)

Is there something different in the impeller DL setup that we're missing for dl opacity peephole? or some other requirement I'm missing?

Reply:
I was finally able to get this running locally and it worked for me with `Grid with alpha SaveLayer on Rows`

Reply:
If you tell me which sub-test types you ran I can look and see why they don't do peephole optimizations. Keep in mind that I created buttons for scenarios that wouldn't work so that we had something to aim for in the future, so not every one of those options will result in opacity peephole success...